### PR TITLE
Port planning_request_adapter_plugins to ROS2

### PR DESCRIPTION
**Pull request by JafarAbdi**

### CODE REVIEW

File: moveit_ros/planning/planning_request_adapters_plugin_description.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<library path="libmoveit_default_planning_request_adapter_plugins">
+<library path="moveit_default_planning_request_adapter_plugins">
```
[the remark this answers is not in the document]
Comment (JafarAbdi) [reply moveit_ros/planning/planning_request_adapters_plugin_description.xml]: Unfortunately, there's no doc for pluginlib for ROS2 -- but after looking at this files it seems like this's correct

- [test_plugins.cpp](https://github.com/ros/pluginlib/blob/dashing/pluginlib/test/test_plugins.cpp)
- [test_package.xml](https://github.com/ros/pluginlib/blob/dashing/pluginlib/test/test_package.xml)
- [test_plugins.xml for dashing](https://github.com/ros/pluginlib/blob/dashing/pluginlib/test/test_plugins.xml)
- [test_plugins.xml for melodic-devel](https://github.com/ros/pluginlib/blob/melodic-devel/pluginlib/test/test_plugins.xml)

File: moveit_ros/planning/planning_request_adapter_plugins/src/add_iterative_spline_parameterization.cpp [hunk 2]
```diff
@@ -38,16 +38,21 @@
...
 class AddIterativeSplineParameterization : public planning_request_adapter::PlanningRequestAdapter
 {
+  rclcpp::Logger LOGGER_ITERATIVE_SPLINE_PARAMETERIZATION =
+      rclcpp::get_logger("moveit_planning_request_adapter_plugins").get_child("add_iterative_spline_parameterization");
+
 public:
   AddIterativeSplineParameterization() : planning_request_adapter::PlanningRequestAdapter()
   {
   }
+  void initialize()
```
[the remark this answers is not in the document]
Comment (JafarAbdi) [reply moveit_ros/planning/planning_request_adapter_plugins/src/add_iterative_spline_parameterization.cpp]: No, just cherry-picking

File: moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp [hunk 3]
```diff
@@ -46,11 +45,16 @@ using namespace trajectory_processing;
 /** @brief This adapter uses the time-optimal trajectory generation method */
 class AddTimeOptimalParameterization : public planning_request_adapter::PlanningRequestAdapter
 {
+  rclcpp::Logger LOGGER_ADD_OPTIMAL_PARAMETERIZATION =
```
Comment (henningkayser) [inline moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp]: Please apply the logging best practices

File: moveit_ros/planning/planning_request_adapter_plugins/src/list.cpp [hunk 4]
```diff
@@ -72,6 +74,6 @@ int main(int argc, char** argv)
       std::cout << " \t\t  " << ad->getDescription() << std::endl;
     std::cout << std::endl << std::endl;
   }
-
+  rclcpp::shutdown();
```
Comment (henningkayser) [inline moveit_ros/planning/planning_request_adapter_plugins/src/list.cpp]: Isn't this redundant?

File: moveit_ros/planning/CMakeLists.txt [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
 
 if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
 find_package(Boost REQUIRED)
 find_package(ament_cmake REQUIRED)
-find_package(moveit_msgs REQUIRED)
-find_package(moveit_ros_perception REQUIRED)
+# find_package(moveit_msgs REQUIRED)
```
Comment (henningkayser) [inline moveit_ros/planning/CMakeLists.txt]: If possible, please keep these enabled

File: moveit_ros/planning/CMakeLists.txt [hunk 6]
```diff
@@ -31,7 +38,7 @@ set(THIS_PACKAGE_INCLUDE_DIRS
     # kinematics_plugin_loader/include
     # robot_model_loader/include
     # constraint_sampler_manager_loader/include
-    planning_pipeline/include
+    # planning_pipeline/include
```
Comment (henningkayser) [inline moveit_ros/planning/CMakeLists.txt]: Doesn't planning_pipeline compile?

File: moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt [hunk 7]
```diff
@@ -1,3 +1,5 @@
+add_compile_options(-Wno-unused-parameter)
```
Comment (henningkayser) [inline moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt]: Please move this into the top-level CMakeLists.txt and add a TODO to remove this. After the sync all these warnings should be resolved anyway.

File: moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp (hunk 3 above)
Comment (henningkayser) [inline moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp]:
```suggestion
  static const rclcpp::Logger LOGGER =
```

File: moveit_ros/planning/CMakeLists.txt (hunk 5 above)
Comment (RoboticsYY) [reply moveit_ros/planning/CMakeLists.txt]: Removed the comment except `moveit_ros_perception`.

File: moveit_ros/planning/CMakeLists.txt (hunk 6 above)
Comment (RoboticsYY) [reply moveit_ros/planning/CMakeLists.txt]: Yes, it compiles. Previously I tried to disable unrelated components. Added it back.

File: moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt (hunk 7 above)
Comment (RoboticsYY) [reply moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt]: Done. Moved it to the top-level CMakeLists.txt. Please check it [here](https://github.com/ros-planning/moveit2/pull/114/files#diff-8a8b87aaf16323bb476afbe7bfd14fbdR9-R15).

File: moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp (hunk 3 above)
Comment (RoboticsYY) [reply moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp]: Changed all logging to their best practices.

**Comment by RoboticsYY** [comment]
> Nice, it compiles, great work! I'm wondering if `COLCON_IGNORE` is actually still needed, the top-level CMakeLists should eliminate all commented subdirectories, and colcon doesn't look for nested packages recursively as far as I now.

`COLCON_IGNORE`s only apply to the ros packages under `moveit_ros` directory. There is no `COLCON_IGNORE` under the subdirectories of `moveit_ros/planning` now.

I run the executable through `ros2 run moveit_ros_planning moveit_list_request_adapter_plugins`. It successfully loaded all the plugins, but ended with a segfault before the return. I'm still trying to figure out the reason.

**Comment by RoboticsYY** [comment]
The segfault seems come from a legacy build of moveit_core. After I removed everything in the /build folder and recompiled the workspace, the segfault disappeared.

File: moveit_ros/planning/planning_request_adapter_plugins/src/fix_start_state_bounds.cpp [hunk 8]
```diff
@@ -49,23 +49,34 @@ class FixStartStateBounds : public planning_request_adapter::PlanningRequestAdap
   static const std::string BOUNDS_PARAM_NAME;
   static const std::string DT_PARAM_NAME;
 
-  FixStartStateBounds() : planning_request_adapter::PlanningRequestAdapter(), nh_("~")
+  void initialize(std::shared_ptr<rclcpp::Node>& node)
```
[the remark this answers is not in the document]
Comment (RoboticsYY) [reply moveit_ros/planning/planning_request_adapter_plugins/src/fix_start_state_bounds.cpp]: @henningkayser I saw there was a [PR](https://github.com/ros-planning/moveit/pull/1530) about this in the original MoveIt repository. Do you mean to still use a private node in the constructor for now, and change the parameter lookup into `initialize` in future?

**Pull request merged**